process_settings.php

```diff
@@ -63,8 +63,10 @@
 
 // Set display errors
 if (isset($settings["display_errors"]) && ($settings["display_errors"])) {
-    error_reporting(E_ALL);
+    // @todo: add setting to allow for deprecated warnings
     ini_set('display_errors', 'on');
+    // show all errors except for E_DEPRECATED warnings
+    error_reporting(E_ALL & ~E_DEPRECATED);
```

Comment:
Why omit the error if we could fix the code?

Reply:
The deprecated messages are warnings (nothing wrong with the code), however they'll print out and break any API responses that require a specific output format.

Reply:
If you'd like to keep all errors and warning messages showing I'll change it back.

Reply:
They are warning that on a next version of PHP the code will not work as the functions are deprecated.
Why wait until it brokes if you can fix it now?